.github/workflows/schedule_tests.yml

```diff
@@ -28,7 +28,9 @@ jobs:
           python-version: ${{ matrix.python-version }}
           cache: 'pip'
           cache-dependency-path: 'tests/requirements/py3.txt'
-      - run: pip install -r tests/requirements/py3.txt -e .
+      - name: Install and upgrade packaging tools
+        run: python -m pip install --upgrade pip setuptools wheel
```

Comment:
I am not sure if or why `setuptools` is needed. pip installs in an isolated environment nowadays and will install the build deps there (via the metadata of the package). Outside of that isolated build env nothing should use setuptools. I am also not exactly sure what benefits the `wheel` package gives us here. Does it help the cache?

Reply:
I see that all of them are installed by default in the runner; so updating does make sense even though aside from pip it might be mostly unneccesary.

Reply:
`wheel` is not installed, see [logs](https://github.com/django/django/runs/5595179319?check_suite_focus=true):
```
Using legacy 'setup.py install' for Pillow, since package 'wheel' is not installed.
Using legacy 'setup.py install' for cffi, since package 'wheel' is not installed.
Using legacy 'setup.py install' for MarkupSafe, since package 'wheel' is not installed.
Using legacy 'setup.py install' for maxminddb, since package 'wheel' is not installed.
Using legacy 'setup.py install' for wrapt, since package 'wheel' is not installed.
```
My main goal what to avoid using `setup.py install` on Windows.

Reply:
:+1: yes then wheel makes certainly sense and I fully agree that using the legacy method is not what we want. So my comment only stands for setuptools -- but if that was already installed it makes sense to upgrade it. I am wondering though if it would make sense to uninstall it instead to maybe uncover some errors.

Reply:
> I am wondering though if it would make sense to uninstall it instead to maybe uncover some errors.

Sounds risky :rofl: